ui/app/AppLayouts/Communities/views/PermissionsView.qml

```diff
@@ -87,14 +88,24 @@ ColumnLayout {
         ]
     }
 
+    StatusListView {
+        id: listView
+        reuseItems: true
+        model: root.permissionsModel
+        spacing: 24
+        Layout.fillWidth: true
+        Layout.fillHeight: !useFullHeight
+        height: useFullHeight ? contentHeight : 0
+        Layout.topMargin: root.topPadding
 
-    Repeater {
-        id: repeater
+        delegate: permissionItemComponent
+    }
 
-        model: root.permissionsModel
+    Component {
+        id: permissionItemComponent
 
-        delegate: PermissionItem {
-            Layout.fillWidth: true
+        PermissionItem {
+            width: root.viewWidth
 
             holdingsListModel: HoldingsSelectionModel {
```

Comment:
This `HoldingsSelectionModel` is sth with potentially big impact on performance (it's kind of proxy model but created when we had no c++ proxies). Probably should be updated and optimized at some point.

Reply:
Yeah, I found this to be the slowest part when joining community / revealing the addresses